### DISCUSSION TITLE
Results not reliable using tokens

### DISCUSSION THREAD

**Discussion by User5679** [microsoft-search/pnp-modern-search#3117] [Q&A]
Hello,

I've come across a behavior that I cannot explain with the PnP Search result Webpart. (**Version used 4.9.0**)
What I am trying to achieve : I have a bunch of News pages in my page library with a category field (fieldtype choice). I want to display on every news the other news with the same category. I created a custom managed property for my category field and I am using the Page token in my query template (example : RefinableString01:{Page.Category}).
The quality of the results varies according to the value of {Page.Category} and seems to be better when this value doesn't contain any space.
Note that I have done the same kind of logic with a multi taxo field and it was working nicely.

**Steps to reproduce :**
1. Create a page content type with a choice field containing simple words and multiple words
2. Create multiple pages from this CT with different values for this field, publish them.
3. Create a managed property mapped on this field and wait for indexation
4. Go on those pages, add the pnp search result webpart and use this query template :
 YourManagedProperty:{Page.YourChoiceFieldInternalName}

**Expected behavior :** 
All the pages with the same field value as the current page should be returned as results.

Now in my page library I have : 

- 6 pages with category = "Europe and beyond" -> Search Query Tool returns 6, PnP Search result returns 1
- 3 pages with category = "Obituaries" -> Search Query Tool returns 3, PnP Search result returns 3
- 2 pages with category = "Au quotidien" -> Search Query Tool returns 2, PnP Search result returns 0

**Search Query Tool result :**
![PnPDiscussion1](https://github.com/microsoft-search/pnp-modern-search/assets/107993013/74d0ec22-fb01-4ac2-bf01-97395769a77a)

**Page details :** 
![GitHubDiscussion3](https://github.com/microsoft-search/pnp-modern-search/assets/107993013/3452e777-69db-424c-8064-e1fed9818958)

**PnP Search result webpart result :** (Trimming duplicates disabled)
![GitHubDiscussion](https://github.com/microsoft-search/pnp-modern-search/assets/107993013/9dcb988d-4040-463a-a5ab-644f8019948e)

**Comment by wobba** [comment]
Did you figure it out?

**Reply by User5679** [reply]
Hello,
Yes in fact I checked other conversations quickly before opening the discussion but I should have paid more attention to this one : #2965 where you gave the answer :). The issue was quite similar and I needed some brackets or quotes for it to show the correct results.
I guess the Search Query Tool handled that by itself by putting quotes in the final query
